library/src/com/google/maps/android/clustering/ClusterManager.java

```diff
@@ -92,13 +93,23 @@ public void setAlgorithm(Algorithm<T> algorithm) {
             if (mAlgorithm != null) {
                 algorithm.addItems(mAlgorithm.getItems());
             }
-            mAlgorithm = new PreCachingAlgorithmDecorator<T>(algorithm);
+
+            mAlgorithm = algorithm;
         } finally {
             mAlgorithmLock.writeLock().unlock();
         }
+
+        if (mAlgorithm instanceof GoogleMap.OnCameraChangeListener) {
+            ((GoogleMap.OnCameraChangeListener) mAlgorithm).onCameraChange(mMap.getCameraPosition());
+        }
+
         cluster();
     }
 
+    public void setClusterOnlyVisibleArea(boolean onlyVisibleArea) {
```

Comment:
I don't think that a user should have to provide an algorithm as well as specifying this property. Can you refactor such that this property is defined on the Algorithm implementation? e.g. add a method to Algorithm like `boolean shouldReclusterOnMapMovement()`